src/kernels/raw/finder/localKernelFinder.node.ts

```diff
@@ -90,38 +79,18 @@ export class LocalKernelFinder implements ILocalKernelFinder, IExtensionSyncActi
             this.disposables
         );
         this.nonPythonKernelFinder.onDidChangeKernels(
-            () => {
-                this.updateCache().then(noop, noop);
-            },
-            this,
-            this.disposables
-        );
-        this.pythonKernelFinder.onDidChangeKernels(
-            () => {
-                this.updateCache().then(noop, noop);
-            },
+            () => this.updateCache().then(noop, noop),
             this,
             this.disposables
         );
+        this.pythonKernelFinder.onDidChangeKernels(() => this.updateCache().then(noop, noop), this, this.disposables);
         this.wasPythonInstalledWhenFetchingControllers = this.extensionChecker.isPythonExtensionInstalled;
     }
 
     private async loadInitialState() {
-        traceVerbose('LocalKernelFinder: load cache');
-        // loading cache, which is resource agnostic
-        const kernelsFromCache = await this.getFromCache();
-        let kernels: LocalKernelConnectionMetadata[] = [];
-
-        this.updateCache().catch(noop);
-
-        if (Array.isArray(kernelsFromCache) && kernelsFromCache.length > 0) {
```

Comment:
We need to cache anything anymore, seems pointless and complicated.
Can add this later if we need this.